lib/connectors/memory.js

```diff
@@ -352,29 +352,49 @@ Memory.prototype.fromDb = function(model, data) {
   if (!data) return null;
   data = deserialize(data);
   var props = this._models[model].properties;
+
   for (var key in data) {
-    var val = data[key];
-    if (val === undefined || val === null) {
-      continue;
-    }
-    if (props[key]) {
-      switch (props[key].type.name) {
-        case 'Date':
-          val = new Date(val.toString().replace(/GMT.*$/, 'GMT'));
-          break;
-        case 'Boolean':
-          val = Boolean(val);
-          break;
-        case 'Number':
-          val = Number(val);
-          break;
-      }
-    }
-    data[key] = val;
+    data[key] = this._castPropertyValue(key, data[key], props);
   }
+
   return data;
 };
 
+Memory.prototype._castPropertyValue = function(prop, val, props) {
+  if (val === undefined || val === null || !props[prop]) {
```

Comment:
See https://github.com/strongloop/loopback-datasource-juggler/pull/1169#discussion_r91471941

> I also recommend to store `props[prop]` in a new variable, so that you don't have to repeat this snippet again and again in the code below.

```js
var propertyType = props[prop]; 
if (val === undefined || val === null || !propertyType)
  return val;
// ...
```